libraries/freertos_plus/standard/freertos_plus_posix/source/FreeRTOS_POSIX_utils.c

```diff
@@ -186,33 +186,40 @@ int UTILS_TimespecAdd( const struct timespec * const x,
                        const struct timespec * const y,
                        struct timespec * const pxResult )
 {
-    int64_t llPartialSec = 0;
+    uint64_t ullPartialSec = 0;
     int iStatus = 0;
 
-    /* Check parameters. */
-    if( ( pxResult == NULL ) || ( x == NULL ) || ( y == NULL ) )
+    /* Check input parameters. None of the parameters should be NULL,
+     * and the input parameters values should be positive. */
+    if( ( pxResult == NULL ) || ( x == NULL ) || ( y == NULL ) ||
+        ( x->tv_sec < 0 ) || ( x->tv_nsec < 0 ) ||
+        ( y->tv_sec < 0 ) || ( y->tv_nsec < 0 ) )
```

Comment:
Is this degenerate unsigned comparison not generating a compiler warning when tv_sec is unsigned?

Reply:
How will we check against negative values otherwise?
I think that the following implementation may also generate warnings:
```
struct timespec temp;
temp.tv_sec = 1;
temp.tv_nsec = 1;

uint8_t isTvSecSigned = (~temp.tv_sec < 0);
uint8_t isTvNsecSigned = (~temp.tv_nsec < 0);
if( ( (isTvSecSigned == 1) && ( ( x->tv_sec < 0) || ( y->tv_sec < 0) ) )  ||
    ( (isTvNSecSigned == 1) && ( ( x->tv_nsec < 0) || ( y->tv_nsec < 0) ) ) )
{
 iStatus = -1;
}
```